app/common/lib/textCalcUtil.js

```diff
@@ -0,0 +1,134 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Immutable = require('immutable')
+
+// Actions
+const appActions = require('../../../js/actions/appActions')
+
+// Constant
+const siteTags = require('../../../js/constants/siteTags')
+
+// Utils
+const tabs = require('../../browser/tabs')
+
+// Styles
+const globalStyles = require('../../renderer/components/styles/global')
+
+const fontSize = parseInt(globalStyles.spacing.bookmarksItemFontSize)
+const fontFamily = globalStyles.defaultFontFamily
+
+const calcText = (item, type) => {
+  const title = type === siteTags.BOOKMARK
+    ? item.get('title') || item.get('location')
+    : item.get('title')
+
+  const param = `
+    (function() {
+      let ctx = document.createElement('canvas').getContext('2d')
+      ctx.font = '${fontSize} ${fontFamily}'
+      const width = ctx.measureText('${title}').width
+      
+      return width
+    })()
+  `
+
+  tabs.executeScriptInBackground(param, (err, url, result) => {
+    if (err) {
+      throw err
+    }
+
+    if (type === siteTags.BOOKMARK) {
+      appActions.onBookmarkWidthChanged(Immutable.fromJS([
+        {
+          key: item.get('key'),
+          parentFolderId: item.get('parentFolderId'),
+          width: result[0]
+        }
+      ]))
+    } else {
+      appActions.onBookmarkFolderWidthChanged(Immutable.fromJS([
+        {
+          key: item.get('key'),
+          parentFolderId: item.get('parentFolderId'),
+          width: result[0]
+        }
+      ]))
+    }
+  })
+}
+const calcTextList = (list) => {
+  const take = 500
+  if (!list.toJS) {
```

Comment:
use immutable helper here for readability pls

Reply:
done